lib/wallet.py

```diff
@@ -90,7 +90,7 @@ def append_utxos_to_inputs(inputs, network, pubkey, txin_type, imax):
     if txin_type == 'p2pkh':
         address = Address.from_pubkey(pubkey)
     else:
-        address = PublicKey.from_pubkey(pubkey)
+        address = P2PKOutput.from_pubkey(pubkey)
```

Comment:
This will break things -- no? P2PKOutput as INPUT will lead to that same bug as before.  Please fix this if that's the case.

Reply:
This is intentional -- note that the following line `address.to_scripthash_hex()` will not work on PublicKey anymore (which is exactly my intent -- generic public keys do not have a script!).

Reply:
Still, I want to test these things fully so no merge yet :)

Reply:
Ok but then go down to like 99.  We have already seen `input['address']`   where `to_ui_string` is not either an address or hex bytes fail in some of the wallet code.